app/javascript/components/rooms/RoomsList.jsx

```diff
@@ -1,14 +1,18 @@
 import React from 'react';
 import {
-  Row, Col, Button, Spinner, Form, FormControl,
+  Row, Col, Button, Form, FormControl,
 } from 'react-bootstrap';
+import Spinner from '../shared/stylings/Spinner';
 import RoomCard from './RoomCard';
 import useRooms from '../../hooks/queries/rooms/useRooms';
-import ButtonLink from '../shared/stylings/buttons/ButtonLink';
+import useCreateRoom from '../../hooks/mutations/rooms/useCreateRoom';
+import RoomPlaceHolder from './RoomPlaceHolder';
 
 export default function RoomsList() {
   const { isLoading, data: rooms } = useRooms();
-  if (isLoading) return <Spinner />;
+  const newRoomData = { name: `Room ${Date.now()}` }; // TODO: amir - change this.
+  const { handleCreateRoom, isLoading: createRoomIsLoading } = useCreateRoom(newRoomData);
```

Comment:
I don't think renaming `isLoading` is necessary here

Reply:
I noticed that we'll have several different `isLoading` states and thought about making renaming them a practice even when it's not necessary.

Reply:
Hmm thats a good point - in that case, I would also rename the `isLoading` in `useRooms`